text/0000-target-bundles.md

```diff
@@ -0,0 +1,245 @@
+- Feature Name: target_bundles
+- Start Date: 2016-08
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Combine distribution of standard libraries and targets into bundles for targeting a particular
+platform. Such bundle is all you need from the Rust side to cross-compile for your target.
+
+The major ideas of this RFC is to:
+
+1. Make JSON targets as full featured as they need to be in order to support specifying targets as
+   custom as they need to be;
+2. Convert current built-in targets to the JSON targets;
+3. Change distribution of libstd to include a corresponding JSON target.
+
+# Motivation
+[motivation]: #motivation
+
+Currently there’s two different ways rustc targets are distributed: built-in targets and custom
+JSON targets. Built-in targets are very inflexible – they cannot be changed without changing and
+recompiling the compiler itself. On the other side of the spectrum, custom JSON targets are easy to
+adjust and adapt, but feature wise are very limited and rarely are suitable for the more uncommon
+use-cases.
+
+We’ve observed a considerable amount of desire by users of the language to customize targets they
+use in the ways currently not supported by our current infrastructure (sans making changes to the
+compiler itself, of course), and noted that the current scheme is not very feasible in the long
+run. This RFC should go a great amount towards fixing the issues.
+
+Then, there also is a strong need to be able to inspect arbitrary parts of target specification,
+regardless of their origin. For example, in a cross-compilation setting, when the crate uses a
+build.rs script, `#[cfg]` variables are for the host, rather than the target. This way the author
+is forced to parse the target triple and figure out particularities of the target on their own, as
+rustc does not provide any way to inspect any of the built-in targets.
```

Comment:
Well, there is `rustc --print cfg`, which does allow quite a bit of inspection for built-in targets (presuming one can execute rustc) [which implies that parsing target triples should never be needed].

That said, I'm in favor of unifying our target specifications.